### PR TITLE
Fix: zk disconnected, dubbo-go panic when subscribe.

### LINKED ISSUE #610: zk disconnected, dubbo-go panic when subscribe.
**What happened**:

When zk disconnected, subscribe may be cause panic.

企业微信截图_3e5feb58-a8a8-4d13-a8bb-dfac43f69ec0

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by wenxuwan**
**What this PR does**:

**Which issue(s) this PR fixes**:
try to fix #610
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by wenxuwan** [comment]
> Travis failing

image

what's the meaning of Travis failing?  As the picture shown, all checks have passed.

**Comment by zouyx** [comment]
> > Travis failing
> 
> image
> 
> what's the meaning of Travis failing? As the picture shown, all checks have passed.

Sorry. File still conflicting

**Comment by wenxuwan** [comment]
> > > Travis failing
> > 
> > 
> > image
> > what's the meaning of Travis failing? As the picture shown, all checks have passed.
> 
> Sorry. File still conflicting

ok, done

File: remoting/zookeeper/listener.go [hunk 1]
```diff
@@ -247,8 +247,13 @@ func (l *ZkEventListener) listenDirEvent(conf *common.URL, zkPath string, listen
 			l.pathMapLock.Lock()
 			l.pathMap[dubboPath] = struct{}{}
 			l.pathMapLock.Unlock()
-
+			//When Zk disconnected, the Conn will be set to nil, so here need check the value of Conn
+			l.client.RLock()
+			if l.client.Conn == nil {
+				break
```
Comment (hxmhlt) [inline remoting/zookeeper/listener.go:253]: Can this case trigger to auto-reconnect?

Comment (zouyx) [reply remoting/zookeeper/listener.go:253]: I have the same question. But travis still failing.

Comment (wenxuwan) [reply remoting/zookeeper/listener.go:253]: In my opinion, there is already have other case for trigger reconnect, so here may be just need stop the resubscribe process and wait reconnect finished.

By the way, the travis failed should not be caused by my changes, the error is:

image

may be you can re-trigger it..

Comment (pantianying) [inline remoting/zookeeper/listener.go:253]: 大哥 这里有死锁吧

Comment (wenxuwan) [reply remoting/zookeeper/listener.go:254]: 是的。。。。

**Pull request merged**